testing/environments/docker/kafka/Dockerfile

```diff
@@ -1,21 +1,21 @@
 FROM debian:buster
 
-ENV KAFKA_HOME /kafka
+ENV KAFKA_HOME=/kafka
 # Controls the hostname advertised within the Docker network, should generally match the container
 # name in a docker-compose file.
-ENV KAFKA_ADVERTISED_HOST kafka
+ENV KAFKA_ADVERTISED_HOST=kafka
 
 ENV KAFKA_LOGS_DIR="/kafka-logs"
-ENV KAFKA_VERSION 2.2.2
-ENV _JAVA_OPTIONS "-Djava.net.preferIPv4Stack=true"
+ENV KAFKA_VERSION=3.5.1
```

Comment:
upgraded kafka version to recommended version as listed here https://docs.aws.amazon.com/msk/latest/developerguide/supported-kafka-versions.html